src/psygnal/_signal.py

```diff
@@ -999,7 +994,8 @@ def emit(
             sd.start()
             return sd
 
-        self._run_emit_loop(args)
+        self._emit_queue.append(args)
+        self._flush_emit_queue()
```

Comment:
`_run_emit_loop` is used in 5 places in codebase. It is intentional to change only this place

Reply:
just an example here for the sake of conversation... not a finished PR